sycl/include/sycl/detail/kernel_desc.hpp

```diff
@@ -151,6 +152,107 @@ template <class KernelNameType> struct KernelInfo {
+    __has_builtin(__builtin_sycl_kernel_param_offset) &&                       \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+    kernel_param_desc_t ParamDesc;
+    ParamDesc.kind =
+        __builtin_sycl_kernel_param_kind(KernelIdentity<KernelNameType>(), I);
+    ParamDesc.info = Result.kind == kind_accessor
+                         ? __builtin_sycl_kernel_param_access_target(
+                               KernelIdentity<KernelNameType>(), I)
+                         : __builtin_sycl_kernel_param_size(
+                               KernelIdentity<KernelNameType>(), I);
+    ParamDesc.offset =
+        __builtin_sycl_kernel_param_offset(KernelIdentity<KernelNameType>(), I);
+    Result.push_back(ParamDesc);
+#else
+    Result.push_back(KernelInfo<KernelNameType>::getParamDesc(I));
+#endif
+  }
+  return Result;
+}
+
+template <typename KernelNameType> constexpr const char *getKernelName() {
+#if __has_builtin(__builtin_sycl_kernel_name) &&                               \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  return __builtin_sycl_kernel_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getName();
+#endif
+}
+
+template <typename KernelNameType> constexpr bool isKernelESIMD() {
+  // TODO Needs a builtin counterpart
+  return KernelInfo<KernelNameType>::isESIMD();
+}
+
+template <typename KernelNameType> constexpr const char *getKernelFileName() {
+#if __has_builtin(__builtin_sycl_kernel_file_name) &&                          \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  return __builtin_sycl_kernel_file_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getFileName();
+#endif
+}
+
+template <typename KernelNameType>
+constexpr const char *getKernelFunctionName() {
+#if __has_builtin(__builtin_sycl_kernel_function_name) &&                      \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  return __builtin_sycl_kernel_function_name(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getFunctionName();
+#endif
+}
+
+template <typename KernelNameType> constexpr unsigned getKernelLineNumber() {
+#if __has_builtin(__builtin_sycl_kernel_line_number) &&                        \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  __builtin_sycl_kernel_line_number(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getLineNumber();
+#endif
+}
+
+template <typename KernelNameType> constexpr unsigned getKernelColumnNumber() {
+#if __has_builtin(__builtin_sycl_kernel_column_number) &&                      \
+    !defined(__INTEL_SYCL_USE_INTEGRATION_HEADERS)
+  __builtin_sycl_kernel_column_number(KernelIdentity<KernelNameType>());
+#else
+  return KernelInfo<KernelNameType>::getColumnNumber();
+#endif
+}
+
+template <typename KernelNameType> constexpr int64_t getKernelSize() {
+#if __has_builtin(__builtin_sycl_kernel_size) &&                               \
```

Comment:
We don't plan to implement this builtin upstream since it is only required for 3rd party compilers. Can this be a TODO for now?

Reply:
Sure, updated.